core/src/test/java/org/elasticsearch/index/mapper/core/DoubleFieldTypeTests.java

```diff
@@ -43,4 +43,9 @@ public void testIsFieldWithinQuery() throws IOException {
         assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(null, randomDouble(), randomDouble(),
                 randomBoolean(), randomBoolean(), null, null));
     }
+
+    public void testValueForSearch() {
```

Comment:
A thought for a follow up: perhaps we could enforce valueForSearch is tested for all field types, by having a test in the base test case which "round trips" a value using indexedValueForSearch and valueForSearch? This would require having an abstract method on the base test case to provide a sample value that is appropriate for the field type.